src/pkg/caendr/caendr/models/datastore/report_entity.py

```diff
@@ -0,0 +1,138 @@
+# Built-ins
+from abc import abstractmethod
+from enum import Enum
+from typing import Union
+
+# Parent class
+from caendr.models.report    import GCPReport
+from caendr.models.datastore import JobEntity, UserOwnedEntity
+
+# Models
+from caendr.models.status    import JobStatus
+from caendr.models.datastore import Container, User
+
+# Services
+from caendr.services.cloud.storage import check_blob_exists
+
+
+
+class ReportEntity(JobEntity, UserOwnedEntity, GCPReport):
+  '''
+    Stores metadata in a Datastore Entity.
+    Manages an entity (metadata) and paths for input/output data
+
+    An Entity subclass that fulfills the Report template class, making it directly usable as a Report in the Job Pipeline.
+
+    Note that the abstract base class GCPReport comes last in the parent class list.
+    This appears to be necessary for the method definitions in the Entity classes
+    to override the abstract methods from GCPReport correctly.
+  '''
+
+  #
+  # Kind
+  #
+
+  @classmethod
+  def get_kind(cls):
+    return cls.kind
+
+
+  #
+  # Instantiation
+  # Route the GCPReport's create & lookup methods to the appropriate Entity methods
+  #
```

Comment:
I think the `Entity` class should also follow this factory method style of instantiation (`lookup` and `create`) -- part of a larger `Entity` redesign project for later